bee-bundle/src/bundle/outgoing_bundle_builder.rs

```diff
@@ -268,27 +403,150 @@ mod tests {
+                .generate_public_key()
+                .unwrap()
+                .trits()
+                .to_owned(),
+        );
+        let address_medium = Address::from_inner_unchecked(
+            WotsPrivateKeyGeneratorBuilder::<Kerl>::default()
+                .security_level(WotsSecurityLevel::Medium)
+                .build()
+                .unwrap()
+                .generate(&seed, 1)
+                .unwrap()
+                .generate_public_key()
+                .unwrap()
+                .trits()
+                .to_owned(),
+        );
+
+        // Transfer
+        bundle_builder.push(default_transaction_builder(0, bundle_size - 1).with_value(Value::from_inner_unchecked(2)));
+
+        // Input
+        bundle_builder.push(
+            default_transaction_builder(1, bundle_size - 1)
+                .with_address(address_low.clone())
+                .with_value(Value::from_inner_unchecked(-1)),
+        );
+        bundle_builder.push(
+            default_transaction_builder(2, bundle_size - 1)
+                .with_address(address_medium.clone())
+                .with_value(Value::from_inner_unchecked(-1))
+        );
+        bundle_builder.push(default_transaction_builder(3, bundle_size - 1).with_address(address_medium.clone()));
+
+        // Build bundle and sign
         let bundle = bundle_builder
             .seal()?
-            .sign()?
+            .sign(
+                &seed,
+                &[
+                    (0, address_low.clone(), WotsSecurityLevel::Low),
+                    (1, address_medium.clone(), WotsSecurityLevel::Medium),
+                ],
+            )?
             .attach_local(Hash::zeros(), Hash::zeros())?
             .build()?;
-
         assert_eq!(bundle.len(), bundle_size);
 
+        // Validate signature
+        let res_low = WotsSignature::<Kerl>::from_buf(bundle.0.get(1).unwrap().payload.to_inner().to_owned())
+            .recover_public_key(bundle.0.get(1).unwrap().bundle.to_inner().as_i8_slice())
+            .unwrap();
+        assert_eq!(address_low.to_inner().as_slice(), res_low.trits());
+
+        let mut signature = TritBuf::<T1B1Buf>::zeros(PAYLOAD_TRIT_LEN * 2);
+        let mut offset = 0;
+        for i in 2..4 {
+            let input = bundle.0.get(i).unwrap();
+            signature.copy_raw_bytes(&input.payload.to_inner().to_owned(), offset, PAYLOAD_TRIT_LEN);
+            offset += PAYLOAD_TRIT_LEN;
+        }
+        let res_medium = WotsSignature::<Kerl>::from_buf(signature)
+            .recover_public_key(bundle.0.get(2).unwrap().bundle.to_inner().as_i8_slice())
+            .unwrap();
+        assert_eq!(address_medium.to_inner().as_slice(), res_medium.trits());
+
         Ok(())
     }
 
+    // TODO Also check to attach if value ?
+    #[test]
+    fn outgoing_bundle_builder_value_test() -> Result<(), OutgoingBundleBuilderError> {
+        // Check each security
+        bundle_builder_signature_check(WotsSecurityLevel::Low)?;
```

Comment:
Does `?` actually make the test to fail if it's `Err` ?

Reply:
will use unwrap() then. We probably want to catch every error and make it panics if happens.